nemo/collections/nlp/data/language_modeling/megatron/dataset_utils.py

```diff
@@ -1200,6 +1194,10 @@ def get_samples_mapping(
 
     # Build the indexed mapping if not exist.
     if torch.distributed.get_rank() == 0 and not os.path.isfile(indexmap_filename):
+        # Fake index mapping if missing
+        if (not hasattr(indexed_dataset, 'doc_idx')) and (not hasattr(indexed_dataset, 'sizes')):
+            make_indexed_dataset_compatibility(indexed_dataset)
```

Comment:
Won't deallocation fail here on ranks that are not rank 0 because they won't have the `doc_idx` and `size attrs`? You are creating these attrs only on rank0, but the deallocation happens on all ranks and doesn't check for these attrs.

Reply:
Deallocation works regardless of compatibility (it assigned None to the desired attributes).

Reply:
Good point. I missed that.